plugins/animate/fire/fire.cpp

```diff
@@ -88,7 +90,46 @@ class FireTransformer : public wf::view_transformer_t
         p.life = 1;
         p.fade = random(0.1, 0.6);
 
-        p.color = {random(0.4, 1), random(0.08, 0.2), random(0.008, 0.018), 1};
+        wf::color_t color_setting = fire_color;
+        bool random_colors = random_fire_color;
+        
+        float r;
+        float g;
+        float b;
+
+        if (!random_colors) {
+            // The calculation here makes the variation lower at darker values
+            float randomize_amount_r = (color_setting.r * 0.857) / 2;
```

Comment:
By the way, where does 0.857 come from?

Reply:
0.857 comes from the original values. I checked what the average for each of the original ranges is, along with how large the difference between them was, and the ratio between them came out to this.

Reply:
Ah, this is why the difference between the old and the new algorithm is negligible .. thanks for putting in the effort!

Reply:
What do you mean by old and new algorithm?